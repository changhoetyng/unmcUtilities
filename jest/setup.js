const rn = require('react-native')

jest.mock('Linking', () => {
    return {
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      openURL: jest.fn(),
      canOpenURL: jest.fn(),
      getInitialURL: jest.fn(),
    }
})

import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';

jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);

module.exports = rn