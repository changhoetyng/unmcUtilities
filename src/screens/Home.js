import React, {Component} from 'react';
import {View, Text, Button, Dimensions} from 'react-native';
import Header from '../assets/header';
import Icon from 'react-native-vector-icons/MaterialIcons';
import HomeOptions from '../assets/homeOptions';

class Home extends Component {
  render() {
    const screenWidth = Dimensions.get('window').width;
    return (
      <View style={{flex: 1}}>
        <Header title="UNMC Utilities" setting={true} />
        <View style={{alignContent: 'center', marginTop: 25}}>
          <HomeOptions
            icon={
              <Icon
                size={screenWidth * 0.18}
                name="sports-handball"
                style={{color: 'white', alignSelf: 'center'}}
              />
            }
            title="Sport Complex"
          />
        </View>
      </View>
    );
  }
}

export default Home;
