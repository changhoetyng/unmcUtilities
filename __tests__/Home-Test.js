import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import Home from '../src/screens/Home';
import renderer from 'react-test-renderer';
const middlewares = [];
const mockStore = configureStore(middlewares);

jest.mock('@react-native-community/async-storage', () =>
  require('@react-native-community/async-storage/jest/async-storage-mock'),
);

const navigation = {
  goBack: jest.fn(),
  navigate: jest.fn(),
};
const currentSelectedMode = jest.fn();
describe('Home', () => {
  const store = mockStore({});
  const rendered = renderer.create(
    <Provider store={store}>
      <Home navigation={navigation} currentSelectedMode={currentSelectedMode} />
    </Provider>,
  );

  beforeEach(() => {
    jest.resetModules();
    jest.resetAllMocks();
    jest.useFakeTimers();
  });

  it('Home Snapshot', () => {
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('Call MyBookingStack if My Booking button is pressed', async () => {
    const button = rendered.root.findByProps({testID: 'myBooking'}).props;
    await button.onPress();
    expect(navigation.navigate).toBeCalledWith('MyBookingStack');
  });

  it('Call BookingStack if roomBooking button is pressed', async () => {
    const button = rendered.root.findByProps({testID: 'roomBooking'}).props;
    await button.onPress();
    expect(navigation.navigate).toBeCalledWith('BookingStack');
  });

  it('Call BookingStack if sportComplex button is pressed', async () => {
    const button = rendered.root.findByProps({testID: 'sportComplex'}).props;
    await button.onPress();
    expect(navigation.navigate).toBeCalledWith('BookingStack');
  });

  it('Call settings if setting button is pressed', async () => {
    const button = rendered.root.findByProps({testID: 'setting'}).props;
    await button.onPress();
    expect(navigation.navigate).toBeCalledWith('Settings');
  });
});
