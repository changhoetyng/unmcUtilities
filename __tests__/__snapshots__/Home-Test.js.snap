// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home Snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#DEE4E8",
      "flex": 1,
    }
  }
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    style={
      Object {
        "zIndex": 1100,
      }
    }
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-around",
          "zIndex": 1000,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "display": "flex",
            "height": 150,
            "justifyContent": "space-around",
            "width": 150,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/loading.gif",
            }
          }
          style={
            Object {
              "height": "70%",
              "width": "70%",
            }
          }
        />
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Loading
        </Text>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#70A9FF",
        "height": "22%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "marginTop": 37.352000000000004,
          },
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "90%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 53.36,
            },
            Object {
              "color": "white",
              "fontWeight": "bold",
            },
          ]
        }
      >
        Home
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="setting"
      >
        <Image
          source={
            Object {
              "uri": "https://cdn1.vectorstock.com/i/thumb-large/77/30/default-avatar-profile-icon-grey-photo-placeholder-vector-17317730.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 46.690000000000005,
              "height": 93.38000000000001,
              "width": 93.38000000000001,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 200.1,
            "marginTop": 40.019999999999996,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "justifyContent": "center",
            "width": "90%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-around",
            "width": "100%",
          }
        }
      >
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "opacity": 1,
              }
            }
            testID="sportComplex"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#888888",
                    "fontSize": 75,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 26.250000000000004,
                }
              }
            >
              Sport Complex
            </Text>
          </View>
        </View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "opacity": 1,
              }
            }
            testID="roomBooking"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#888888",
                    "fontSize": 75,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 26.250000000000004,
                }
              }
            >
              Room Booking
            </Text>
          </View>
        </View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "opacity": 1,
              }
            }
            testID="myBooking"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#888888",
                    "fontSize": 75,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 26.250000000000004,
                }
              }
            >
              My Booking
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 106.72,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 45.75,
            },
            Object {
              "color": "black",
              "fontWeight": "bold",
              "marginBottom": 3,
            },
          ]
        }
      >
        Announcement
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 25,
              },
              Object {
                "color": "#70A9FF",
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    collapsable={false}
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "width": "90%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
